File: modules/agent/src/mocks/repositories/productRepo.ts
import { readFileSync } from 'fs';
import { dirname, join } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const DATA_PATH = join(__dirname, '..', 'data');

interface Product {
  sku: string;
  name: string;
  unit_cost: number;
}

let products: Map<string, Product> | null = null;

const loadData = (): void => {
  if (products) return;

  const data = JSON.parse(readFileSync(join(DATA_PATH, 'products.json'), 'utf-8'));
  products = new Map(data.products.map((p: Product) => [p.sku, p]));
};

export const getAllProducts = (): Product[] => {
  loadData();
  return Array.from(products!.values());
};

export const findProductBySku = (sku: string): Product | undefined => {
  loadData();
  return products!.get(sku);
};

File: modules/agent/src/agent/mocks/repositories/product_repo.py
import json
from pathlib import Path

DATA_PATH = Path(__file__).parent.parent / "data"

_products: dict[str, dict] = {}


def _load_data():
    global _products
    if _products:
        return
    with open(DATA_PATH / "products.json") as f:
        _products = {p["sku"]: p for p in json.load(f)["products"]}


def get_all_products() -> list[dict]:
    _load_data()
    return list(_products.values())


def find_product_by_sku(sku: str) -> dict | None:
    _load_data()
    return _products.get(sku)
